### PR TITLE
refactor(wasm): modularize wasm crate logic

### PR DESCRIPTION
**Pull request by TalDerei**
References #3431 and https://github.com/penumbra-zone/penumbra/issues/3400.

### CODE REVIEW

File: crates/wasm/src/build.rs
Comment (hdevalence) [inline crates/wasm/src/build.rs:41]: Why do we need to match here? Can't we call the existing ActionPlan::build_unauth method?

Comment (TalDerei) [reply crates/wasm/src/build.rs:41]: good point, noticed that as well. I'll make the change.

**Comment by TalDerei** [comment]
@hdevalence looking at `build_action` method again, I'm realizing it doesn't need to take the [transaction_plan](https://github.com/penumbra-zone/penumbra/blob/wasm-builder/crates/wasm/src/build.rs#L21) field. It's deserializing an entire `TransactionPlan` just to access the `memo_plan` which can simply be passed in from the web-side.

**Comment by hdevalence** [comment]
Yeah, I thought the reason for that was that it would be simpler to only pass one object over FFI into wasm. Even if it's a little inefficient, that's fine if the code becomes simpler, as long as it's not the dominant performance factor.

File: crates/wasm/src/build.rs
Comment (grod220) [inline crates/wasm/src/build.rs:32]: Can we just annotate `WitnessData` so we get the auto-try-into stuff?

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(try_from = "pb::WitnessData", into = "pb::WitnessData")]
pub struct WitnessData {
```

File: crates/wasm/src/build.rs
Comment (grod220) [inline crates/wasm/src/build.rs:42]: Does it have to be an Option? or can we pass it like:

```rust
serde_wasm_bindgen::to_value(&action)?
```

File: crates/wasm/src/wasm_planner.rs
Comment (grod220) [inline crates/wasm/src/wasm_planner.rs:82]: for wasm_bindgen, it's not necessary adding annotations for struct methods. The rendered class will include them automatically.

**Review by grod220 — COMMENTED** [review]
For the wasm-compilation script, left comments in the other PR 👍

File: crates/wasm/src/build.rs
Comment (TalDerei) [reply crates/wasm/src/build.rs:32]: We can, references https://github.com/penumbra-zone/penumbra/issues/3400. 

I modified the witness as `let witness: WitnessData = serde_wasm_bindgen::from_value(witness_data)?`

File: crates/wasm/src/build.rs
Comment (TalDerei) [reply crates/wasm/src/build.rs:43]: changed to simply `result`

File: crates/wasm/src/build.rs
Comment (TalDerei) [reply crates/wasm/src/build.rs:42]: Good catch, I'm not sure why this was marked as an option. The call to `build_unauth` expects a known `ActionPlan` to be specified. I made the change.

File: crates/wasm/src/wasm_planner.rs
Comment (TalDerei) [reply crates/wasm/src/wasm_planner.rs:82]: removed

**Comment by TalDerei** [comment]
This PR is blocked until #3429 is merged.

**Review by grod220 — APPROVED** [review]

**Pull request merged**
